src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/DefaultConfiguration.java

```diff
@@ -31,8 +31,9 @@ public Node mapNodeLabel(final String label) {
     }
 
     @Override
-    public String unmapNodeLabel(String label) {
-        return label.replaceAll(NS + NODELABEL, "");
+    public String unmapNodeLabel(final Node label) {
+        final String text = label.isURI()? label.getURI() : label.getLiteralValue().toString();
+        return text.replaceAll(NS + NODELABEL, "");
```

Comment:
I don't think that this is correct. This function should be the inverse of `mapNodeLabel`. In this sense, it shouldn't be able to handle `Node` objects that represent literals and, in fact, it shouldn't even be able to handle all URIs, but only those that can be produced by `mapNodeLabel`. Hence, it should throw an IllegalArgumentException for every other type of `Node` object (which should be exactly all the `Node` objects for which `mapsToLabel` returns `false`).